google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/stub/metrics/BuiltinMetricsTracer.java

```diff
@@ -268,6 +273,12 @@ public void grpcMessageSent() {
     grpcMessageSentDelay.set(attemptTimer.elapsed(TimeUnit.NANOSECONDS));
   }
 
+  @Override
+  public void setRemainingDeadline(long deadline) {
```

Comment:
I don't think this is right :( `deadline - timeElapsed` could be negative if there are retries. I think for now we can just log and publish the deadline. It should also be published per attempt. This way we'll know if the super short deadline we see on the server side is from customer's settings or from retry attempts.

Reply:
Done!